File: esercizi_java/test/compito_python_4/Gestione.py
import sys
import argparse

from Corso import Corso
from Laboratorio import Laboratorio
from Studente import Studente
from Teoria import Teoria
from Libretto import Libretto


class Gestione:
    def __init__(self, arg):
        self._arg = arg

    def main(self):
        corsi = []
        studenti = []
        codProg = {}

        # punto 1
        try:
            f = open("corsi.txt", "r")
            line = f.readline().strip()
            while(line != ''):
                tok = line.split()
                cod = int(tok[0])
                tipo = tok[1]
                line = f.readline().strip()
                nomecorso = line
                line = f.readline().strip()
                nomedoc = line
                line = f.readline().strip()
                codProg[cod] = nomecorso
                if(tipo == "teoria"):
                    tok = line.split()
                    codaula = int(tok[0])
                    oreset = int(tok[1])
                    orelez = float(tok[2])
                    t = Teoria(cod, nomecorso, nomedoc,
                               codaula, oreset, orelez)
                    corsi.append(t)

                else:
                    nomelab = line
                    line = f.readline().strip()
                    nomeass = line
                    line = f.readline().strip()
                    numpos = int(line)
                    l = Laboratorio(cod, nomecorso, nomedoc,
                                    nomelab, nomeass, numpos)
                    corsi.append(l)
                line = f.readline().strip()
                line = f.readline().strip()
            f.close()

        except IOError:
            print("IO error finded")
        except:
            print("error finded", sys.exc_info()[0])
            raise

        # punto 2

        try:
            f = open("studenti.txt", "r")
            line = f.readline().strip()
            while(line != ''):
                cod = int(line)
                line = f.readline().strip()
                nome = line
                s = Studente(cod, nome)
                studenti.append(s)
                line = f.readline().strip()
                while(line != '' and line != ""):
                    tok = line.split()
                    codcor = int(tok[0])
                    voto = tok[1]

                    # aggiunta di addCorso che non funziona come in java
                    l = Libretto(codcor, voto)
                    s.addCorso(l)

                    line = f.readline().strip()
                line = f.readline().strip()
            f.close()

        except IOError:
            print("IOError findend")
        except:
            print("Error findend", sys.exc_info()[0])
            raise

        # punto 3
        print("nome, codice, docente, tipo, aula, ore sett., ore/lez., lab., assistente, postazioni")
        for c in corsi:
            print(c.toString())

        # punto 4
        print("\n")
        for s in studenti:
            print(s.toString())

        # punto 5
        print("\n")
        a = 5
        for s in studenti:
            if a == s._cod:
                print(s._nome + "\t"+str(s.max()._voto) +
                      "\t"+str(codProg[s.max()._cod]))


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument('matricola', action='store')
    #matr = parser.parse_args()
    matr = 10
    g = Gestione(matr)
    g.main()

File: esercizi_java/NetBeansProjects/Universita/src/universita/Gestione.java
package universita;

import java.util.LinkedList;
import java.util.List;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;
import java.io.IOException;
import java.io.BufferedReader;
import java.io.FileReader;


/**
 *
 * @author tobiasacchetto
 */
public class Gestione {

    static List<Corso> corsi = new LinkedList<Corso>();
    static List<Studente> studenti = new LinkedList<Studente>();
    static Map<Integer, Corso> codProg= new HashMap<Integer, Corso>();
    
    
    public static void main(String[] args) {
       
        //punto 1
        
        try{
        BufferedReader br = new BufferedReader(new FileReader("corsi.txt"));
        String line = br.readLine();
        while(line!= null){
            StringTokenizer tok = new StringTokenizer(line);
            int cod = Integer.parseInt(tok.nextToken());
            String tipo = tok.nextToken();
            line = br.readLine();
            String nomecorso = line;
            line = br.readLine();
            String nomedoc = line;
            line = br.readLine();
            if(tipo.equals("teoria")){          
                tok = new StringTokenizer(line);
                int codaula = Integer.parseInt(tok.nextToken());
                int oreset = Integer.parseInt(tok.nextToken());
                double orelez = Double.parseDouble(tok.nextToken());
                Teoria t = new Teoria(cod,nomecorso,nomedoc,codaula,oreset,orelez);
                corsi.add(t);
                codProg.put(cod,t);
                

            }else{
                String nomelab = line;
                line = br.readLine();
                String nomeass = line;
                line = br.readLine();
                int numpos = Integer.parseInt(line);
                Laboratorio l = new Laboratorio(cod,nomecorso,nomedoc,nomelab,nomeass,numpos);
                corsi.add(l);
                codProg.put(cod,l);
                
                
            }
            line = br.readLine();
            line = br.readLine();
        }
        br.close();
        
        }catch(IOException e){
            System.err.print(e);
        }catch(Exception e){
            System.err.print(e);
        }
        
        
        //punto 2
        
        try{
            BufferedReader br = new BufferedReader(new FileReader("studenti.txt"));
            String line = br.readLine();
            while(line!=null){
                int cod = Integer.parseInt(line);
                line = br.readLine();
                String nome = line;
                Studente s = new Studente(cod,nome);
                studenti.add(s);
                line = br.readLine();
                while(line!=null && !line.equals("") ){ //devi mettere prima la null e poi l'equal senno non funziona
                    StringTokenizer tok = new StringTokenizer(line);
                    int codcor = Integer.parseInt(tok.nextToken());
                    String voto= tok.nextToken();
                    s.addCorso(codProg.get(codcor), voto);
                    line = br.readLine();
                }
                if(line!=null)
                    line = br.readLine();
            }
            br.close();
        }catch(IOException e){
            System.err.print(e);
        }catch(Exception e){
            System.err.print(e);
        }
        
        
        //punto 3
        
        System.out.println("nome, codice, docente, tipo, aula, ore sett., ore/lez., lab., assistente, postazioni");
        for(Corso c: corsi){
            System.out.println(c);
        }
        
        
        //punto 4
        System.out.println();
        for(Studente s: studenti){
            System.out.println(s);
        }
        
        //punto 5
        System.out.println();
        int b=Integer.parseInt(args[0]);
        for(Studente s: studenti){
            if(b==s.cod){
                System.out.println(s.nome+"\t"+s.max());
            }
        }
    }
    
}
